package/MDAnalysis/lib/pkdtree.py

```diff
@@ -127,22 +130,22 @@ def set_coords(self, coords, cutoff=None):
 
         """
         # If no cutoff distance is provided but PBC aware
-        if self.pbc and (cutoff is None):
-            raise RuntimeError('Provide a cutoff distance'
-                               ' with tree.set_coords(...)')
 
         # set coords dtype to float32
         # augment coordinates will work only with float32
         coords = np.asarray(coords, dtype=np.float32)
 
         if self.pbc:
-            self.cutoff = cutoff
+            if self.cutoff is None:
```

Comment:
Here, the original test is not about `self.cutoff`, it is instead about `cutoff` as provided as an argument to the function. This seems to be why the tests fail.

```suggestion
            if cutoff is None:
```

It is subtle and got me puzzled for a while!